src/saveload/order_sl.cpp

```diff
@@ -102,35 +102,61 @@ Order UnpackOldOrder(uint16_t packed)
 	return order;
 }
 
+/** Temporary storage for conversion from old order pool. */
+static std::vector<OldOrderSaveLoadItem> _old_order_saveload_pool;
+
+/**
+ * Clear all old orders.
+ */
+void ClearOldOrders()
+{
+	_old_order_saveload_pool.clear();
+	_old_order_saveload_pool.shrink_to_fit();
+}
+
+/**
+ * Get a pointer to an old order with the given reference index.
+ * @param ref_index Reference index (one-based) to get.
+ * @return Pointer to old order, or nullptr if not present.
+ */
+OldOrderSaveLoadItem *GetOldOrder(size_t ref_index)
+{
+	if (ref_index == 0) return nullptr;
+	assert(ref_index <= _old_order_saveload_pool.size());
+	return &_old_order_saveload_pool[ref_index - 1];
+}
+
+/**
+ * Allocate an old order with the given pool index.
+ * @param pool_index Pool index (zero-based) to allocate.
+ * @return Reference to allocated old order.
+ */
+OldOrderSaveLoadItem &AllocateOldOrder(size_t pool_index)
+{
+	assert(pool_index < UINT32_MAX);
+	if (pool_index >= _old_order_saveload_pool.size()) _old_order_saveload_pool.resize(pool_index + 1);
+	return _old_order_saveload_pool[pool_index];
+}
+
 SaveLoadTable GetOrderDescription()
 {
 	static const SaveLoad _order_desc[] = {
-		     SLE_VAR(Order, type,           SLE_UINT8),
-		     SLE_VAR(Order, flags,          SLE_UINT8),
-		     SLE_VAR(Order, dest,           SLE_UINT16),
-		     SLE_REF(Order, next,           REF_ORDER),
-		 SLE_CONDVAR(Order, refit_cargo,    SLE_UINT8,   SLV_36, SL_MAX_VERSION),
-		 SLE_CONDVAR(Order, wait_time,      SLE_UINT16,  SLV_67, SL_MAX_VERSION),
-		 SLE_CONDVAR(Order, travel_time,    SLE_UINT16,  SLV_67, SL_MAX_VERSION),
-		 SLE_CONDVAR(Order, max_speed,      SLE_UINT16, SLV_172, SL_MAX_VERSION),
+		     SLE_VARNAME(OldOrderSaveLoadItem, order.type,  "type",  SLE_UINT8),
+		     SLE_VARNAME(OldOrderSaveLoadItem, order.flags, "flags", SLE_UINT8),
+		     SLE_VARNAME(OldOrderSaveLoadItem, order.dest,  "dest",  SLE_UINT16),
+		 SLE_CONDVARNAME(OldOrderSaveLoadItem, next,        "next",  SLE_FILE_U16 | SLE_VAR_U32, SL_MIN_VERSION, SLV_69),
+		 SLE_CONDVARNAME(OldOrderSaveLoadItem, next,        "next",  SLE_UINT32,                 SLV_69, SL_MAX_VERSION),
+		 SLE_CONDVARNAME(OldOrderSaveLoadItem, order.refit_cargo, "refit_cargo", SLE_UINT8,   SLV_36, SL_MAX_VERSION),
+		 SLE_CONDVARNAME(OldOrderSaveLoadItem, order.wait_time,   "wait_time",   SLE_UINT16,  SLV_67, SL_MAX_VERSION),
+		 SLE_CONDVARNAME(OldOrderSaveLoadItem, order.travel_time, "travel_time", SLE_UINT16,  SLV_67, SL_MAX_VERSION),
+		 SLE_CONDVARNAME(OldOrderSaveLoadItem, order.max_speed,   "max_speed",   SLE_UINT16, SLV_172, SL_MAX_VERSION),
```

Comment:
I think making them conditional is not needed. The ORDR chunk will not exist in the newer save versions as its read only.

Reply:
They're conditional because the start version is different.